libs/s25main/ingameWindows/IngameWindow.cpp

```diff
@@ -162,26 +163,23 @@ void IngameWindow::MouseLeftUp(const MouseCoords& mc)
 {
     isMoving = false;
 
-    const std::array<Rect, 2> rec = {GetCloseButtonBounds(), GetMinimizeButtonBounds()};
-
-    for(unsigned i = 0; i < 2; ++i)
+    for(const auto id : helpers::enumRange<IwButtonId>())
     {
-        buttonState[i] = ButtonState::Up;
+        buttonStates_[id] = ButtonState::Up;
 
-        if((i == 0 && closeBehavior_ == CloseBehavior::Custom) // no close button
-           || (i == 1 && isModal_))                            // modal windows cannot be minimized
-        {
+        if((id == IwButtonId::Close && closeBehavior_ == CloseBehavior::Custom) // no close button
+           || (id != IwButtonId::Close && isModal_))                            // modal windows cannot be minimized
```

Comment:
Wouldn't this be more in line with the comment?
```suggestion
           || (id == IwButtonId::Minimize && isModal_))                            // modal windows cannot be minimized
```

Reply:
No. But that's only because you haven't seen the commit from #1606. "Not the close button" makes sense when there're 3 in total.